source/coffee/image-picker.coffee

```diff
@@ -26,14 +26,11 @@ sanitized_options = (opts) ->
   jQuery.extend(default_options, opts)
 
 both_array_are_equal = (a,b) ->
-  if (!a || !b) || (a.length != b.length)
-    return false
-  b = b[..]
-  for x in a
-    i = b.indexOf x
-    if i < 0
-      return false
-    b.splice(i, 1)
+  return false if (!a || !b) || (a.length != b.length)
```

Comment:
Ah good point. We could simply clone the arrays. 

```
var a = [3,2,1]
var b = [3,2,4]
function sortArrays(a, b) {
    a = a.slice(0)
    b = b.slice(0)
    a.sort()
    b.sort()
    console.log(a, b)
}

console.log(a, b)
>>>> [3, 2, 1] [3, 2, 4]
sortArrays(a, b)
>>>> [1, 2, 3] [2, 3, 4]
console.log(a, b)
>>>> [3, 2, 1] [3, 2, 4]
```

If we know that the arrays are always already sorted we could skip sorting in this function.